### PR TITLE
Pixelorigin

### LINKED ISSUE #402: Strange behavior on map edges
**Describe the bug:**
What happened vs. what did you expect to happen?

When image is scaled above map edges it expand on the sides.
When is moved beyond the edges it behaves strangely.

Expected: image stops scaling/moving when it touches borders or scales/moves normally.

**Reproduce the behavior:**
 Steps to reproduce the behavior. (ex. 1. Go to... 2. Click on... 3. See error...)

You can see this behavior here:
https://jsfiddle.net/adriankwiat/fz9vbghq/33/

**Browser, version, and operating system:**
 - Browser [e.g. chrome, safari] -
 - Version [e.g. 22] -
 - Platform: [e.g. iOS, windows] -

### PR DESCRIPTION
**Pull request by sashadev-sky**
Fixes #402  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updates
* [ ] @mention the original creator of the issue in a comment below for help or for a review

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

======= 
changes: 
- updated debug/pixelorigin.html to use one of our distortableImageOverlay's instead of a marker
- bounds are now saved on the image corner and updated appropriately - one of the pending things wanted to implement to fix downstream MK logic (https://github.com/publiclab/mapknitter/pull/959)
    - note that `setBounds` calls `_reset`, so we no longer use that method ourselves.
- updated dragging logic to use `setCorners()` as well, also mentioned as useful in https://github.com/publiclab/mapknitter/pull/959
- bump to `0.7.8`
- update docs

pending:

=======

### CODE REVIEW

**Comment by sashadev-sky** [comment]
@jywarren @rexagod here is an explanation of this one because its a little tricky - 

the typical `L.Draggable` leaflet object is allowed to just leave the map:

typical

Since we overrode this method, instead we get a result like the one shown in #402:
63525312-c5741580-c4fd-11e9-9503-6e67e9621636

To fix this I update `setCorners` with a conditional that uses a formula based on the information from Leaflet docs that 
image

which came out like this:
```JS
   for (var k in latlngObj) {
      if ((zoom === 0 && (map.project(latlngObj[k]).y < 2 || map.project(latlngObj[k]).y >= 255)) ||
          (zoom !== 0 && (map.project(latlngObj[k]).y / zoom < 2 || map.project(latlngObj[k]).y / Math.pow(2, zoom) >= 255))
      ) {
        // calling reset / update w/ the same corners bc it prevents a marker flicker for rotate
        this.setBounds(L.latLngBounds(this.getCorners()));
        this.fire('update');
        return;
      }
    }
```

2 is for the top, 255 is for the bottom. Made it less than 256 because 256 would trigger the distortions. 

This results in:
now

**Comment by sashadev-sky** [comment]
let me know your thoughts and ill make corresponding doc updates if you like it!

File: src/edit/ScaleHandle.js [hunk 1]
```diff
@@ -2,44 +2,46 @@ L.ScaleHandle = L.EditHandle.extend({
...
      */
 
     if (!edgeMinWidth) { edgeMinWidth = 50; } /* just in case */
-    var corner1 = map.latLngToLayerPoint(overlay.getCorner(0)),
-		corner2 = map.latLngToLayerPoint(overlay.getCorner(1)),
-		w = Math.abs(corner1.x - corner2.x),
-		h = Math.abs(corner1.y - corner2.y),
-		distance = Math.sqrt(w * w + h * h);
+    var corner1 = map.latLngToLayerPoint(overlay.getCorner(0));
+    var corner2 = map.latLngToLayerPoint(overlay.getCorner(1));
```
Comment (jywarren) [inline src/edit/ScaleHandle.js]: These commas + var always cause trouble. I'm cool with this syntax!

Comment (sashadev-sky) [reply src/edit/ScaleHandle.js]: @jywarren were switching to ES6 soon so I think either way its ok but its what we decided to go with in #389

**Comment by sashadev-sky** [comment]
@jywarren merging here first because its already ready.

**Pull request merged**